helm/prometheus-rules/templates/platform/atlas/alerting-rules/loki.rules.yml

```diff
@@ -135,3 +135,39 @@ spec:
         severity: page
         team: atlas
         topic: observability
+  - name: loki.compactor
+    rules:
+    - alert: LokiCompactorFailedCompaction
+      annotations:
+        dashboard: loki-retention/loki-retention
+        description: 'Loki compactor has been failing compactions for more than 2 hours since last compaction.'
+        opsrecipe: loki#lokicompactorfailedcompaction
+      # This alert checks if Loki's the last successful compaction run is older than 2 hours
+      expr: (time() - (loki_boltdb_shipper_compact_tables_operation_last_successful_run_timestamp_seconds > 0) > 60 * 60 * 2)
+      for: 1h
+      labels:
+        area: platform
+        cancel_if_cluster_status_creating: "true"
+        cancel_if_cluster_status_deleting: "true"
+        cancel_if_cluster_status_updating: "true"
+        cancel_if_outside_working_hours: "true"
+        severity: page
+        team: atlas
+        topic: observability
+    - alert: LokiCompactorFailedCompaction
+      annotations:
+        dashboard: loki-retention/loki-retention
+        description: 'Loki compactor has been failing compactions for more than 2 hours since start-up.'
```

Comment:
I don't understand why we need this second alert.
What's the difference between "the compactor already ran successfully, in the past but failed for the past 2 hours" and "the compactor was never successful for the last 2 hours" ? Why do we care about the difference?

Reply:
The issue is that when loki starts up (restarts are included), it's possible it does not need to run any compaction in the first few hours of it's lifetime. In that case the value of  `loki_boltdb_shipper_compact_tables_operation_last_successful_run_timestamp_seconds` will be 0.

That means that we will receive an alert after 1h (instead of 2) because the alert query would become `(time() - 0) > 2h)` => `time() > 2h` and time() will always greater than 2 hours.

Reply:
oh, understood :bulb: